File: fib.cpp
#include<iostream>
#include<string>
#include<stdlib.h>
#include<map>

using namespace std;

const int RECURSION = 0;
const int ITERATIVE = 1;
map <string, int> methodMapping = {
									{"r", RECURSION},
									{"i", ITERATIVE}
								};

long fibRec(int n) 
{
	if (n < 2) {
		return n;
	} else {
		return fibRec(n - 1) + fibRec(n - 2);
	}
}

long fibIter(int n) 
{
	long first = 1;
	long second = 1;
	for (int i = 2; i < n; i++) {
		long next = first + second;
		first = second;
		second = next;
	}
	return second;
}


int main(int argc, char * argv[]) 
{
	int n = atoi(argv[1]);
	string method = argv[2];
	long fib = 0;
	switch (methodMapping[method]) 
	{
		case RECURSION:
		{
			fib = fibRec(n);
			break;
		}
		case ITERATIVE:
		{
			fib = fibIter(n);
			break;
		}
	}
	cout << "C++ done " << method <<"\n" << fib << endl;
	return 0;
}

File: fib.js
#!/usr/bin/jjs
function fibRec(n) {
  if (n < 2) {
    return n;
  } else {
    return fibRec(n - 1) + fibRec(n - 2);
  }
}

function fibIter(n){
	var first = 1;
	var second = 1;
	for (var i = 2; i < n; i++) {
		var next = first + second;
		first = second;
		second = next;
	}
	return second;
}

n = arguments[0];
method = arguments[1];
var fib = 0;
switch (method) {
	case 'r':
		fib = fibRec(n);
		break;
	case 'i':
				fib = fibIter(n);
		break;
}
print("JavaScript done by " + method + ":\n" + 	fib);
